zarr/storage.py

```diff
@@ -2073,6 +2083,7 @@ def __init__(
         self.mode = mode
         self.open = open
         self.write_lock = write_lock
+        self.write_mutex: Union[Lock, NoLock]
```

Comment:
this is not a blocker, but we should look into making `NoLock` an instance of `Lock`